09_employment/compute_metrics_wage_ratio_2022_subgroup_new.do

```diff
@@ -0,0 +1,200 @@
+
+/* replace 0 ratio with missing and replace data quality as missing */
+replace average_to_living_wage_ratio = "NA" if average_to_living_wage_ratio == "."
+replace wage_ratio_quality = 3 if average_to_living_wage_ratio == "NA"
+
+save "wage_ratio_final_2022_subgroup.dta",replace
+
+
+keep state county year average_to_living_wage_ratio wage_ratio_quality industry_type
+
+/* rename final variable here */
+
+*rename ratio_average_to_living_wage average_to_living_wage_ratio
+rename wage_ratio_quality ratio_living_wage_quality
+rename industry_type subgroup
+
+generate subgroup_type = 1 if subgroup == 1
+replace subgroup_type = 2 if subgroup != 1
+
+label define subg_type_lab 1 "all" 2 "industry"
+
+label values subgroup_type subg_type_lab
+
+/* export files */
+
+order year state county subgroup_type subgroup average_to_living_wage_ratio ratio_living_wage_quality
+
+export delimited using metrics_wage_ratio_2022_subgroup.csv, replace
+
+save "wage_ratio_final_2022_subgroup.dta",replace
+
+** add in county name and state name
+global crosswalk "C:\Users\KMartinchek\Documents\umf-2024\mobility-from-poverty\geographic-crosswalks\data"
+
+import delimited using "$crosswalk\county-populations.csv", stringcols(2 3 4 5) varnames(1) clear
+
+merge 1:m year state county using "wage_ratio_final_2022_subgroup.dta"
+
+** correct CT counties -- this an issue since they switched to county equilvalents in 2022 -- and this data (QECW) uses the old (2020) FIPS
+replace state_name = "Connecticut" if state == "09"
+replace state_name = "Alaska" if state == "02"
+
+replace county_name = "Fairfield County" if state == "09" & county == "001"
+replace county_name = "Hartford County" if state == "09" & county == "003"
+replace county_name = "Litchfield County" if state == "09" & county == "005"
+replace county_name = "Middlesex County" if state == "09" & county == "007"
+replace county_name = "New Haven County" if state == "09" & county == "009"
+replace county_name = "New London County" if state == "09" & county == "011"
+replace county_name = "Tolland County" if state == "09" & county == "013"
+replace county_name = "Windham County" if state == "09" & county == "015"
+replace county_name = "Valdez-Cordova Census Area" if state == "02" & county == "261"
+
+drop if _merge == 1 & state != "09"
+
+*** also want to drop old counties and impute missing values 
+generate flag = 1 if state == "09" & average_to_living_wage_ratio == "NA"
+drop if flag == 1
+
+*** make missing
+replace average_to_living_wage_ratio = "NA" if state == "09"
+replace ratio_living_wage_quality = 3 if state == "09"
+
+/* export final dataset */
+
+keep if year == 2022
+
+keep year state county state_name county_name subgroup_type subgroup average_to_living_wage_ratio ratio_living_wage_quality
+
+order year state county state_name county_name subgroup_type subgroup  average_to_living_wage_ratio ratio_living_wage_quality
+
+export delimited using metrics_wage_ratio_2022_subgroup.csv, replace
+
```

Comment:
sort file by year, state, and county -- currently out of order

Reply:
The "order" command orders them properly-- and when I open the csv, I see the correct order-- so I am not sure this is an issue.